src/server/game/Config/GameConfig.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _GAME_CONFIG_H
+#define _GAME_CONFIG_H
+
+#include "Common.h"
+#include <unordered_map>
+
+enum class GameConfigType : uint8
+{
+    GAME_CONFIG_TYPE_BOOL,
+    GAME_CONFIG_TYPE_INT,
+    GAME_CONFIG_TYPE_FLOAT,
+    GAME_CONFIG_TYPE_STRING,
+
+    GAME_CONFIG_TYPE_UNKNOWN
+};
+
+class TC_GAME_API GameConfig
+{
+public:
+    static GameConfig* instance();
+
+    void Load();
+
+    // Get config options
+    bool GetBoolConfig(std::string const& optionName);
```

Comment:
maybe these functions could accept a default value like the sWorld ones ?

Reply:
I'm not sure this is necessary because the system works as follows, if you don't set a custom value, the function will return the default value